File: objc/examples/html_extraction.m
// html_extraction — build a PDF from Markdown, then render it back to HTML.
// Shared-scenario regression example. Returns non-zero on assertion failure.
#import "POXPdfOxide.h"
#import <Foundation/Foundation.h>

int main(void) {
    @autoreleasepool {
        NSError* err = nil;
        POXPdf* pdf = [POXPdf fromMarkdown:@"# Hello pdf_oxide\n\nThis is a "
                                           @"**Objective-C** regression example.\n"
                                     error:&err];
        if (!pdf) {
            fprintf(stderr, "error: %s\n", err.localizedDescription.UTF8String);
            return 1;
        }

        POXDocument* doc = [POXDocument openFromBytes:[pdf toBytesWithError:&err]
                                                error:&err];
        if (!doc) {
            fprintf(stderr, "error: %s\n", err.localizedDescription.UTF8String);
            return 1;
        }

        NSString* html = [doc toHtmlAllWithError:&err];
        if (!html) {
            fprintf(stderr, "error: %s\n", err.localizedDescription.UTF8String);
            return 1;
        }
        printf("%s\n", html.UTF8String);

        if (![html containsString:@"<"]) {
            fprintf(stderr, "assertion failed: html missing '<'\n");
            return 1;
        }
        if (![html containsString:@"pdf_oxide"]) {
            fprintf(stderr, "assertion failed: html missing 'pdf_oxide'\n");
            return 1;
        }
        printf("HTML OK\n");
        return 0;
    }
}

File: zig/examples/html_extraction.zig
// html_extraction — build a PDF from Markdown, open it, render HTML.
//
// Shared regression scenario (mirrored across language bindings). Exits
// non-zero on any failed assertion; prints "HTML OK" on success.
const std = @import("std");
const pdf_oxide = @import("pdf_oxide");

pub fn main() !void {
    const a = std.heap.page_allocator;

    var pdf = try pdf_oxide.Pdf.fromMarkdown(
        "# Hello pdf_oxide\n\nThis is a **Zig** regression example.\n",
    );
    defer pdf.deinit();

    const bytes = try pdf.toBytes(a);
    defer a.free(bytes);

    var doc = try pdf_oxide.Document.openFromBytes(bytes);
    defer doc.deinit();

    const html = try doc.toHtmlAll(a);
    defer a.free(html);

    var buf: [4096]u8 = undefined;
    var fw = std.fs.File.stdout().writer(&buf);
    const stdout = &fw.interface;

    try stdout.print("--- html (all) ---\n{s}\n", .{html});

    if (std.mem.indexOfScalar(u8, html, '<') == null) {
        try stdout.print("assertion failed: html does not contain '<'\n", .{});
        try stdout.flush();
        std.process.exit(1);
    }
    if (std.mem.indexOf(u8, html, "pdf_oxide") == null) {
        try stdout.print("assertion failed: html does not contain 'pdf_oxide'\n", .{});
        try stdout.flush();
        std.process.exit(1);
    }

    try stdout.print("HTML OK\n", .{});
    try stdout.flush();
}
